.github/workflows/build_docs.yml

```diff
@@ -0,0 +1,56 @@
+name: Build and deploy docs
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.10']
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install Dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install .[extra]
+        pip install .[doc]
+    - name: Build documentation
+      run: |
+        cd doc
+        make -C . html-no-examples
+    - name: Deploy Devs
+      if: success() && github.ref == 'refs/heads/master'
+      uses: JamesIves/github-pages-deploy-action@v4
+      with:
+        ssh-key: ${{ secrets.ACTIONS_DEPLOY_KEY }}
+        repository-name: dipy/preview-html
+        folder: ./doc/_build/html
+        target-folder: dev
+    - name: Update last release
+      if: success() && github.ref == 'refs/heads/master'
```

Comment:
this seems to be the same as above so no difference between dev and release. I think I misunderstood something.

Reply:
The target folder here is the one corresponding to the last release. It is hard-coded, though:
https://github.com/dipy/dipy/pull/3295/files#diff-4dfe935b4852393622c2b414783d1d11e1a868fd8485709124d90208fa61ec8bR55

However, I now see what you mean, and I agree, it does not make sense here; a separate action should take care of the deployment when a release event is triggered.